File: API/Entities/Member.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Entities;

public class Member
{
    public string Id { get; set; } = null!;
    public DateOnly DateOfBirth { get; set; }
    public string? ImageUrl { get; set; }
    public required string DisplayName { get; set; } = null!;
    public DateTime Created { get; set; } = DateTime.UtcNow;
    public DateTime LastActive { get; set; } = DateTime.UtcNow;
    public required string Gender { get; set; } = null!;
    public string? Description { get; set; }
    public required string City { get; set; } = null!;
    public string Country { get; set; } = null!;

    // Navigation property (one-to-one relationship with AppUser)
    //
    // [ForeignKey(nameof(Id))]
    // - Tells EF that Member.Id is also the foreign key
    // - This creates a shared primary key relationship
    // - Member.Id == AppUser.Id
    [ForeignKey(nameof(Id))]
    public AppUser User { get; set; } = null!;
    public ICollection<Photo> Photos { get; set; } = [];

}

File: client/src/interface/member.ts
export interface Member {
  id: string;
  dateOfBirth: string; // Returned as string from API
  imageUrl?: string; // Optional (user may not have image)
  displayName: string;
  created: string; // ISO date string
  lastActive: string; // ISO date string
  gender: string;
  description?: string; // Optional (not required at signup)
  city: string;
  country: string;
}

export interface EditableMember {
  displayName: string;
  description?: string;
  city: string;
  country: string;
}
